### ISSUE TITLE
you have to give name kwargs to parameters/variables in jupyter notebooks

### ISSUE THREAD

**Issue by Jhsmit** [tBuLi/symfit#60]
Havent tested ipython but in jupyter you cant make paramters without using the name kwargs. This makes sense but it could be nice to give a useful error message to the user urging him to specify 'name'.

**Comment by Pitje06** [comment]
IIRC this happens in all interactive sessions, and is due to problems with the introspection module not working properly there. 
I don't know if we can catch this specific error, or should submit it somewhere upstream.

**Comment by Jhsmit** [comment]
It seems to be possible to detect if python is in interactive mode:

```python
def is_interactive():
    import __main__ as main
    return not hasattr(main, '__file__')
```
http://stackoverflow.com/questions/15411967/how-can-i-check-if-code-is-executed-in-the-ipython-notebook

**Comment by Pitje06** [comment]
I can't reproduce the original issue in either Ipython 5.1.0, python 3.5.2, nor in a jupyter notebook 4.4.2. Can you still reproduce it?

**Comment by Jhsmit** [comment]
Nope seems to work for me in jupyter notebook. Somebody must have fixed it.

**Comment by tBuLi** [comment]
If there is one design choice I regret it's using inspect to guess the names. It introduces more problems than it solves.

I am definitely considering removing it, but still having the name be optional for backwards compatibility, and throwing a deprecation warning. If no name is provided, make it a sympy.dummy parameter, just like I already do with dependent variables if no name is provided.

**Comment by Jhsmit** [comment]
Black magic is not to be used lightly. 

If a user doesnt bother with explicitly giving a name its not likely he will explicitly call it. However I'm wondering what a `print(result)` will look like when no name is known. Numbered parameters by order of creation?

**Comment by pckroon** [comment]
Closed by #140
